src/careamics/dataset_ng/patch_extractor/image_stack/czi_image_stack.py

```diff
@@ -0,0 +1,297 @@
+import re
+from collections.abc import Sequence
+from pathlib import Path
+from typing import Any, Literal
+
+import numpy as np
+from numpy.typing import NDArray
+from pylibCZIrw.czi import CziReader, Rectangle, open_czi
+
+
+class CziImageStack:
+    """
+    A class for extracting patches from an image stack that is stored as a CZI file.
+
```

Comment:
Could you add a few words on the use of the bounding rectangle? It may not makes sense, but if it is defined then it will be used, otherwise the whole range is used?

Reply:
I'm not sure what you mean since bounding rectangle is not mentioned in this docstring. I elaborated the documentation of the `scene`  argument and hope that clarifies it.

Reply:
Very clear, thanks!